optimum/intel/openvino/modeling.py

```diff
@@ -129,8 +129,12 @@ def to(self, device: str):
         Use the specified `device` for inference. For example: "cpu" or "gpu". `device` can
         be in upper or lower case. To speed up first inference, call `.compile()` after `.to()`.
         """
-        self._device = str(device).upper()
-        self.request = None
+        if isinstance(device, str):
+            self._device = device.upper()
+            self.request = None
+        else:
+            logger.warning(f"device must be of type {str} but got {type(device)} instead")
```

Comment:
The issue is fixed (thank you!), I tested that with this PR these pipelines execute on GPU again, but this warning shows every  time:

![image](https://github.com/huggingface/optimum-intel/assets/77325899/64aa5b43-f7ee-4d57-9661-96835a1b4e32)